### PR TITLE
Escape invalid characters in generated dataflow job name

### PR DESCRIPTION
**Pull request by ThePletch**

### CODE REVIEW

File: orchestration/templates/stage-data.yaml [hunk 1]
```diff
@@ -83,7 +83,7 @@ spec:
           {{- with .Values.dataflow }}
           - --project={{ .project }}
           - --region={{ .region }}
-          - --jobName=hca-stage-metadata-{{ $stagingPrefix }}
+          - --jobName=hca-stage-metadata-{{ regexReplaceAll "[^-a-z0-9]+" $stagingPrefix "-" }}
```
Comment (aherbst-broad) [inline orchestration/templates/stage-data.yaml:86]: Where does regexReplaceAll come from?

Comment (ThePletch) [reply orchestration/templates/stage-data.yaml:86]: revisiting this, it comes from helm, which i mistakenly thought was accessible from argo

**Review by raaidbroad — APPROVED** [review]
Regex looks sane, again per usual argo things we don't have a good way to know if this will truly work till we try so let 'er rip imo

File: orchestration/templates/stage-data.yaml (hunk 1 above)
Comment (ThePletch) [reply orchestration/templates/stage-data.yaml]: update: helm is accessible from argo. the syntax here was incorrect for function invocations, however. those need to be wrapped in hyphens in addition to double braces. updated.

**Review by aherbst-broad — APPROVED** [review]

**Pull request merged**